feat: Implement socket connection in Setting component for real-time updates

choiseona opened the pull request:
## 📂 작업 내용

closes #100 

- [x] 방 생성시 기본 설정 maxPlayers:5, totalRounds:5, drawTime:30
- [x] 설정 maxPlayers:(4,5), totalRound:(3,5), drawTime:(15, 20, 25, 30)
- [x] 방장은 설정을 변경할 수 있음
- [x] 방장이 아닌 참가자는 설정을 확인할 수 있음

## 💡 자세한 설명

- updateSetting 요청은 gameSocketStore에 구현하였습니다.
- settingsUpdated 응답은 useGameSocket에 구현하였습니다.
- 방 생성시 기본적으로 maxPlayers:5, totalRounds:5, drawTime:30로 설정됩니다.
- 설정은 maxPlayers:(4,5), totalRound:(3,5), drawTime:(15, 20, 25, 30) 가능합니다. 
- type이 participant면(방장x) 설정을 변경할 수 없도록 하였습니다.
- type이 host이면(방장o) 설정을 변경할 수 있도록 하였습니다. 


## 📗 참고 자료 & 구현 결과 (선택)

- 실시간 반영 
![socket1](https://github.com/user-attachments/assets/06336fd6-e614-4e69-bbca-03140c769fba)

- 플레이어수 -> 최대 플레이어 수 레이아웃 
![layout](https://github.com/user-attachments/assets/8d7311df-9da4-459f-ade2-36635d47e0f1)


## 📢 리뷰 요구 사항 (선택)

## 🚩 후속 작업 (선택)

## ✅ 셀프 체크리스트

- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요? (`main`이 아닙니다.)
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 테스트는 잘 통과했나요?
- [x] 불필요한 코드는 제거했나요?

choiseona committed:
feat: Implement updateSettings socket emission for updating game settings

choiseona committed:
feat: Implement settingsUpdated handler to update room settings in game state

choiseona committed:
feat: Connect Setting component to socket for real-time updates

choiseona committed:
feat: Update options for drawTime setting in ROOM_SETTINGS

choiseona committed:
fix: Change type from 'host' to 'participant' in Setting component

rhino-ty commented on client/src/components/setting/Setting.tsx:
```diff
@@ -12,25 +13,32 @@ interface RoomSettingItem {
-  { label: '제한 시간', key: 'drawTime', options: [15, 30] },
-  { label: '픽셀 수', key: 'maxPixels', options: [300, 500] },
+  { label: '라운드 수', key: 'totalRounds', options: [3, 5] },
+  { label: '플레이어 수', key: 'maxPlayers', options: [4, 5] },
```
'최대' 플레이어 수로 작성해주세요! 디자인이 깨지지 않나 확인도 같이 부탁드립니다!

choiseona committed:
Refactor: Update room settings handling

- Replaced the initialization of selectedValues with Partial<RoomSettings> to allow optional fields.
- Added logic to set selectedValues when 
oomSettings changes.
- Adjusted useEffect to handle settings update only when selectedValues or 	ype changes.

choiseona committed:
design: Update max-width for better layout
